qubes-dom0-packagev2.yml: use new base config, handle existing compon…

krystian-hebel opened the pull request:
…ents

krystian-hebel committed:
qubes-dom0-packagev2.yml: use new base config, handle existing components

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
qubes-dom0-packagev2.yml: anchor Ubuntu version to 24.04

Qubes Builder requires datetime.UTC from Python, which was added in 3.11.
ubuntu-latest (currently ubuntu-22.04) uses 3.10.12 by default.

ubuntu-latest temporarily moved to 24.04, but was quickly reverted back
to 22.04.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
qubes-dom0-packagev2.yml: print contents of files created by workflow

... except for 'rel', which always contains '1' and thus doesn't provide
any information useful for debugging.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel commented:
CI run with these changes: https://github.com/TrenchBoot/xen/actions/runs/11684066822/job/32534536652?pr=19

SergiiDmytruk commented:
> CI run with these changes: https://github.com/TrenchBoot/xen/actions/runs/11684066822/job/32534536652?pr=19

This surprised me with the contents of the `version` file:
```
  0+a9f4275561fe671c85994ca74f51756e9ddd7a04
  a9f4275561fe671c85994ca74f51756e9ddd7a04
```

I guess it still works, but maybe we should add `| head -1` there to not debug weird behaviour at some later point?
https://github.com/TrenchBoot/.github/blob/802b07632e46bebc01ca50f22e9eb89275d7dd50/.github/workflows/qubes-dom0-packagev2.yml#L111

krystian-hebel committed:
qubes-dom0-packagev2.yml: use only first line of git show-ref output

'git show-ref -s "$branch_name"' outputs two identical references.
While this produces file that gets accepted by current version of Qubes
builder, this isn't the expected format and it may break in the future.
Use only the first line to avoid breakage in the future.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel commented:
> I guess it still works, but maybe we should add `| head -1` there to not debug weird behaviour at some later point?

Added in https://github.com/TrenchBoot/.github/pull/13/commits/26a1020d5584b536df483ee5ac3bb55b30aeef8b
